devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -225,8 +249,20 @@ public void execute() throws MojoFailureException, MojoExecutionException {
         if (!sourceDir.isDirectory()) {
             getLog().warn("The project's sources directory does not exist " + sourceDir);
         }
+        //we check to see if there was a compile (or later) goal before this plugin
```

Comment:
I think you only need to check if `compile` was executed? Not sure if you can `install` or `deploy` without the `compile` phase is invoked

Reply:
Just realized these are the goals set from the `mvn` command. Looks ok